packages/playwright-test/src/webServer.ts

```diff
@@ -38,18 +38,25 @@ export class WebServer {
     this._isAvailable = getIsAvailableFunction(config, reporter.onStdErr?.bind(reporter));
   }
 
-  public static async create(config: WebServerConfig, reporter: Reporter): Promise<WebServer> {
-    const webServer = new WebServer(config, reporter);
+  public async start(options: { setBaseURL: boolean } = { setBaseURL: true }) {
     try {
-      await webServer._startProcess();
-      await webServer._waitForProcess();
-      return webServer;
+      await this._startProcess();
+      await this._waitForProcess();
+      if (options.setBaseURL && this.config.port !== undefined)
```

Comment:
I'd do this vice versa - make `WebServer` a part of plugin, and instantiate the plugin in response to `config.webServer` property. This way you'll get to set the `baseURL` through `configure()` phase and remove this env var.

Reply:
I like that. Latest patchset has it. It does create some weirdness for logging/reporters since configure runs before we have a reporter, but that can all be mitigated with the resolution to https://github.com/microsoft/playwright/pull/13779#discussion_r860139204